rust_src/src/multibyte.rs

```diff
@@ -136,13 +138,37 @@ impl<'a> Iterator for LispStringRefIterator<'a> {
     }
 }
 
+impl<'a> Iterator for LispStringRefCharIterator<'a> {
+    type Item = Codepoint;
+
+    fn next(&mut self) -> Option<Codepoint> {
+        self.0.next().map(|result| result.0)
+    }
+}
+
+impl<'a> Iterator for LispStringRefIndexIterator<'a> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        self.0.next().map(|result| result.1)
+    }
+}
+
 impl LispStringRef {
     pub fn iter(&self) -> LispStringRefIterator {
         LispStringRefIterator {
             string_ref: self,
             cur: 0,
         }
     }
+
+    pub fn chars(&self) -> LispStringRefCharIterator {
+        LispStringRefCharIterator(self.iter())
+    }
+
+    pub fn char_indices(&self) -> LispStringRefIndexIterator {
```

Comment:
Sorry to nitpick again, but actually `char_indices` returns the `(index, char)` tuple - you just need to rename `iter`.